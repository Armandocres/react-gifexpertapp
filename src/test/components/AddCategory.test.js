import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Probando Addcategory', () => {
  const setCattegory = jest.fn();
  let wrapper = shallow(<AddCategory  setCategories={setCattegory} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory  setCategories={setCattegory} />);
  });

  test('Snapshot de Addcategory', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de cambiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'Hola mundo';

    input.simulate('change', { target: { value: value }
    });
  });

  test('No debe de postear la informacion onSubmit', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} } );

    expect(setCattegory).not.toHaveBeenCalled();
  })

  test('Debe llamar al setcategpries y limpiar la caja de texto', () => {
    const value = 'Hola mundo';
    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault(){} });
    expect(setCattegory).toHaveBeenCalled();
    expect(wrapper.find('input').prop('value')).toBe('');
  })

});