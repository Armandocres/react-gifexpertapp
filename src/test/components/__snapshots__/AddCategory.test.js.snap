// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando Addcategory Snapshot de Addcategory 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="gift"
  >
    Busca un gift
  </label>
  <input
    id="gift"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
